packages/frontend/src/providers/Web3Provider.tsx

```diff
@@ -11,19 +11,22 @@ import { WagmiConfig, configureChains, createClient } from 'wagmi';
 import { publicProvider } from 'wagmi/providers/public';
 import { mainnet } from 'wagmi/chains';
 import { infuraProvider } from 'wagmi/providers/infura';
+import { alchemyProvider } from 'wagmi/providers/alchemy';
 
 const { chains, provider } = configureChains(
   [mainnet],
   [
+    alchemyProvider({ apiKey: 'F2zZplZomYztFDGt01AjNSXPtFfM3APh' }),
+    publicProvider(),
     infuraProvider({
       apiKey: '1f04715c0a10492483047083a5c3edd2',
     }),
```

Comment:
Keep the public provider as the last option
```suggestion
    alchemyProvider({ apiKey: 'F2zZplZomYztFDGt01AjNSXPtFfM3APh' }),
    infuraProvider({
      apiKey: '1f04715c0a10492483047083a5c3edd2',
    }),
    publicProvider(),
```

Reply:
@kkatusic NEVER EVER put API keys directly into the code. Use env variables here and add them to `.env.example` so other devs can know which env variables are necessary to run the project.

Reply:
@kkatusic the project is very simple and using only Alchemy as a provider is ok, what was the reasoning behind also addind Infura?

Reply:
@kkatusic sorry, I mismatched the name, I meant to ask why you added the alchemy provider if only infura was being used.

Reply:
@kkatusic the scope of the change is to fix the mobile issue with rainbowkit, not a migration of providers, please make each PR address a specific problem. If you think that changing the provider is relevant here, please add a new PR with this scope and justify the change. It's ok to keep with the current provider, especially because that doesn't affect the issue we're discussing on this PR.